src/query/storage/converter.go

```diff
@@ -279,7 +280,12 @@ func iteratorToTsSeries(
 	datapoints := make(ts.Datapoints, 0, initRawFetchAllocSize)
 	for iter.Next() {
 		dp, _, _ := iter.Current()
-		datapoints = append(datapoints, ts.Datapoint{Timestamp: dp.Timestamp, Value: dp.Value})
+		// NB: re-apply the offset to every datapoint here to move the decompressed
+		// series back to the correct timeframe.
```

Comment:
Hm, does it feel strange doing this in the `query/storage` package?

Reply:
This is more a side effect of where this converter thing is; it should probably be in an m3 specific one since it's converting series iterators to points